### PR TITLE
add new version of cxxopt package

### PR DESCRIPTION
**Pull request by hubert-lee**
I have completed remote test.
https://travis-ci.org/hubert-lee/hunter/builds/386545254
https://ci.appveyor.com/project/hubert-lee/hunter/build/1.0.3

cxxopts 1.2.0/1.3.0 is not hunterized. so I excluded these version.
If you want to add above versions, allow permission hunter-package repository.
(https://github.com/hunter-packages/cxxopts)
if so, I can release cxxopts-1.2.0/1.3.0 in hunter-package and I will give pull-request again.

### CODE REVIEW

File: examples/cxxopts/CMakeLists.txt [hunk 1]
```diff
@@ -9,8 +9,11 @@ include("../common.cmake")
 
 project(download-cxxopts)
 
+
 hunter_add_package(cxxopts)
 find_package(cxxopts CONFIG REQUIRED)
+string(REPLACE "." ";" VERSION_LIST ${HUNTER_cxxopts_VERSION})
+list(GET VERSION_LIST 0 CXXOPTS_MAJOR)
```
Comment (ruslo) [inline examples/cxxopts/CMakeLists.txt]: I don't see the usage of `CXXOPTS_MAJOR` variable, can it be removed?

File: examples/cxxopts/v1.3.0 [hunk 2]
```diff
@@ -0,0 +1 @@
+./v1.x.x
```
Comment (ruslo) [inline examples/cxxopts/v1.3.0]: Unused?

**Comment by ruslo** [comment]
```
    I've tested this package remotely and have excluded all broken builds.
    Here is the links to the Travis/AppVeyor with status "All passed":
        https://ci.appveyor.com/project//hunter/build/
        https://travis-ci.org//hunter/builds/

    This update will break few old toolchains.
    They are excluded in this pull request: https://github.com/ingenue/hunter/pull/
```

Links are broken, second part is not relevant as far as I can see. Please do clean up.

> cxxopts 1.2.0/1.3.0 is not hunterized. so I excluded these version.
> If you want to add above versions, allow permission hunter-package repository

`cxxopts` package don't have dependencies (at least in configuration which is used) so the term [hunterization](https://docs.hunter.sh/en/latest/reference/terminology/hunterize.html) is not really applied here. Fork https://github.com/hunter-packages/cxxopts was created to apply few CMake tweaks:
* https://github.com/hunter-packages/cxxopts/commits/hunter

Since install configuration was added to upstream actually fork is not needed anymore and can be removed. I will schedule removing `cxxopt 1.0.0-p0` from Hunter and removing of https://github.com/hunter-packages/cxxopts fork.

File: examples/cxxopts/CMakeLists.txt (hunk 1 above)
Comment (hubert-lee) [reply examples/cxxopts/CMakeLists.txt]: Oh, I forgot to remove that variable. 
I will remove it.

File: examples/cxxopts/v1.3.0 (hunk 2 above)
Comment (hubert-lee) [reply examples/cxxopts/v1.3.0]: `v1.3.0` `v1.4.0` `v1.4.1` `v1.4.2` `v1.4.3` `v1.4.4` 
Above files are symnolic link and points to `v1.x.x` directory.
`v1.x.x` `v2.0.0` `v2.1.0` are directory that contain `example.cpp` source file.
`example.cpp` does not support lower versions. So, I divided into versions.

File: examples/cxxopts/CMakeLists.txt [hunk 3]
```diff
@@ -8,9 +8,8 @@ cmake_minimum_required(VERSION 3.0)
 include("../common.cmake")
 
 project(download-cxxopts)
-
 hunter_add_package(cxxopts)
 find_package(cxxopts CONFIG REQUIRED)
-
-add_executable(foo example.cpp)
+string(REPLACE "." ";" VERSION_LIST ${HUNTER_cxxopts_VERSION})
```
Comment (ruslo) [inline examples/cxxopts/CMakeLists.txt]: `VERSION_LIST` is not used too

File: examples/cxxopts/v1.3.0 (hunk 2 above)
Comment (ruslo) [reply examples/cxxopts/v1.3.0]: Probably more clean solution will be to do:
```cmake
if(HUNTER_cxxopts_VERSION VERSION_LESS "2.0.0")
  set(source_dir "v1.x.x")
else()
  set(source_dir "v${HUNTER_cxxopts_VERSION}")
endif()

add_executable(foo "${source_dir}/example.cpp")
```

**Comment by hubert-lee** [comment]
Your solution is very clear. so, I modify code. Thank you.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.4053
* https://travis-ci.org/ingenue/hunter/builds/387632387?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://travis-ci.org/ingenue/hunter/builds/387635232?utm_source=github_status&utm_medium=notification
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.4054

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.22.2

**Pull request merged**